articles/kotlin-ranges/index.md

```diff
@@ -0,0 +1,134 @@
+
+```bash
+1,2,3,4,5,
+```
+
+This loop goes from 1 to 5. That's not the only trick ranges can do. We can also create ranges using characters. We can create a range that starts from a specific letter in the alphabet to another. This can be from `a` to `e`. You can go ahead and call the `printCharRange` function.
+
+The expected output is:
+
+```bash
+a,b,c,d,e,
+```
+It was not easy to do this in java. When implemented it led to lots of boilerplate and dirty code.
+
+```Java
+  String letters = "abcdefghijklmopqrstuvwxyz'";
+  for (char c : letters.toCharArray()) {
+      System.out.printf(c);
+  }
+```
+
+First we create a string of the characters. Then we convert it to a CharArray where using `forEach`, we get each of the characters. With Kotlin ranges, this is reduced to only a few lines with fewer requirements.
+
+### 2. To Create Reverse Loops
+It has always been a challenge to create a reverse loop, as a developer would have interchange values the old way. This could lead to confusion within the code and it would take time to debug.
+
+An example of a reverse loop in Java would be:
+
+```Java
+for (int i = 5; i >= 0; i--) {}
+```
+
+Ranges have a function called `downTo` that can be used to create a reverse progression. We can use it to loop from the end value backwards to the start point.
+
+Once you run the `reverseNumRange`, the output should be:
+
+```bash
+5,4,3,2,1,
+```
+
+The same applies to the character range. It loops downwards to the start character. The function `reverseCharRange` does this for us.
+
+```bash
+e,d,c,b,a,
+```
+
+### 3. To Generate Random Numbers
+Sometimes you may want to generate random numbers within certain limits. This is virtually impossible in the old loop. A good option is to use the `Math` class and call the `random` function. This however is limiting to the values it returns.
+
+It returns double values ranging from 0.0 to 1.0. In addition to that, you would need to add more calculations. It may be confusing to a developer who is not very experienced in Java.
+
+For example, to generate random numbers between 1 to 10:
+
+```Java
+int max = 10;
+int min = 1;
+int range = max - min + 1;
+int random = (Math.random() * range) + min;
+```
+
+With ranges, it is as simple as creating the desired range and calling the `random` method. This makes it easy to get random numbers in any range. In the `getRandomInRange` function, we are creating a range from 1 to 5 then getting a random number from it. We add the value to the output through Kotlin's [String interpolation](https://kotlincompact.com/string-interpolation.html). You can add expressions as well as variables in strings.
+
+### 4. To Check For Values in Ranges
+Sometimes you may need to check if a value lies inside a particular range. For instance, after picking an image via `intents`, we can check the dimensions. Suppose, we need the image dimensions to lie in a specific range, we can use the range operator.
```

Comment:
> picking an image via intents

It would add value if you could elaborate on this.

Reply:
I have added a link to the android official docs on picking images using intents. I have also mentioned `android` to give context to my example.